api/src/main/java/org/apache/iceberg/Schema.java

```diff
@@ -215,7 +215,7 @@ public StructType asStruct() {
   public Set<String> identifierFieldNames() {
     return identifierFieldIds()
             .stream()
-            .map(id -> findField(id).name())
+            .map(id -> lazyIdToName().get(id))
```

Comment:
this is another bug not really related to the discussion, trying to just merge it here instead of having another 1 line PR.